constructor/nsis/main.nsi.tmpl

```diff
@@ -1202,13 +1230,26 @@ Section "Uninstall"
     # In case the last command fails, run the slow method to remove leftover
     RMDir /r /REBOOTOK "$INSTDIR"
 
-    DeleteRegKey SHCTX "${UNINSTREG}"
+    ${If} $INSTALLER_NAME_FULL != ""
+        DeleteRegKey SHCTX "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$INSTALLER_NAME_FULL"
+    ${EndIf}
+
     # If Anaconda was registered as the official Python for this version,
     # remove it from the registry
-    ReadRegStr $0 SHCTX "Software\Python\PythonCore\${PY_VER}\InstallPath" ""
-    ${If} $0 == "$INSTDIR"
-        DeleteRegKey SHCTX "Software\Python\PythonCore\${PY_VER}"
-    ${EndIf}
+    StrCpy $R0 "SOFTWARE\Python\PythonCore"
```

Comment:
This block is easier to understand, I think. Basically we don't want to assume that the `base` Python is still the same version. However, is this updated when `conda` updates the `python` package in `base`? I don't remember seeing code for this 🤔 It would still be the same path, so it would work, but maybe the metadata is outdated when it comes to the reported version in the registry.

Reply:
This also prevents issues in cross-installer usage, as described in my other comment.

I do not think that `conda` updates the registry when python is updated, which it should probably do.